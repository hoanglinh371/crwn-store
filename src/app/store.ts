import { configureStore, ThunkAction, Action } from '@reduxjs/toolkit';

import cartReducer from '../features/cart/redux/cart.slice';
import categoryReducer from '../features/category/redux/category.slice';
import productReducer from '../features/product/redux/product.slice';

export const store = configureStore({
  reducer: {
    productReducer,
    cartReducer,
    categoryReducer,
  },
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
