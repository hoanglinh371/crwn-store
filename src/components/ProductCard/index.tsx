import React from 'react';
import classNames from 'classnames/bind';
import { useAppDispatch } from 'src/app/hooks';

import { Product } from 'src/features/product/redux/product.types';
import { addProduct } from 'src/features/cart/redux/cart.slice';
import styles from './ProductCard.module.scss';

import Button from '../Button';

interface Props {
  product: Product;
}

const cx = classNames.bind(styles);

const ProductCard: React.FC<Props> = ({ product }) => {
  const { name, price, imageUrl } = product;
  const dispatch = useAppDispatch();

  const handleAddToCart = () => dispatch(addProduct(product));

  return (
    <div className={cx('container')}>
      <img src={imageUrl} alt={name} />
      <div className={cx('footer')}>
        <span aria-label='name' className={cx('name')}>
          {name}
        </span>
        <span className={cx('price')}>${price}</span>
      </div>
      <Button type='button' inverted onClick={handleAddToCart}>
        Add To Cart
      </Button>
    </div>
  );
};

export default ProductCard;
