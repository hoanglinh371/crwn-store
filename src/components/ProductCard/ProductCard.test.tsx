import { cleanup, render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';

import { store } from 'src/app/store';
import { Product } from 'src/interfaces/product.interface';
import ProductCard from '.';

const mockProduct: Product = {
  id: 1,
  name: 'Red Hat',
  imageUrl: 'red-hat-image',
  price: 99,
};

describe('ProductCard component', () => {
  afterEach(cleanup);

  it('Should_MatchSnapshot', () => {
    const { container } = render(
      <Provider store={store}>
        <ProductCard product={mockProduct} />
      </Provider>,
    );
    expect(container).toMatchSnapshot();
  });

  it('Should_HaveNameProduct', () => {
    render(
      <Provider store={store}>
        <ProductCard product={mockProduct} />
      </Provider>,
    );
    expect(screen.getByLabelText('name')).toHaveTextContent('Red Hat');
  });
});
