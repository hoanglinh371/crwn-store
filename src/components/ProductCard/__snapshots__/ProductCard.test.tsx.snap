// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCard component Should_MatchSnapshot 1`] = `
<div>
  <div
    class="container"
  >
    <img
      alt="Red Hat"
      src="red-hat-image"
    />
    <div
      class="footer"
    >
      <span
        aria-label="name"
        class="name"
      >
        Red Hat
      </span>
      <span
        class="price"
      >
        $
        99
      </span>
    </div>
    <button
      aria-label="button"
      class="container inverted"
      type="button"
    >
      Add To Cart
    </button>
  </div>
</div>
`;
